src/operator/nn/dnnl/dnnl_where.cc

```diff
@@ -34,7 +34,16 @@ namespace op {
 
 // Support for https://oneapi-src.github.io/oneDNN/v2.6/dev_guide_binary.html
 bool SupportDNNLWhere(const std::vector<NDArray>& inputs) {
-  return SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::Mixed>(inputs);
+  if (inputs[0].dtype() == mshadow::kBool) {
+    // oneDNN natively doesn't support bool data type, however this operator was written
+    // to allow using bool datatype for 'condition' tensor - data will be treated as uint8
+    bool bool_support = SupportDNNLShape<1, 12>(inputs[0].shape());
+    return bool_support &&
+           SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::AllSame>({inputs[1], inputs[2]});
```

Comment:
```suggestion
    return SupportDNNLShape<1, 12>(inputs[0].shape()) &&
           SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::AllSame>({inputs[1], inputs[2]});
```
If you insist on keeping the variable please make it const.